src/main/java/org/fenixedu/learning/servlets/FenixEduLearningContextListener.java

```diff
@@ -148,11 +149,11 @@ private static void handleThesisProposalApproval(DomainObjectEvent<Thesis> event
                     BundleUtil.getLocalizedString(Bundle.MESSAGING, "thesis.announcement.subject", thesis.getStudent()
                             .getPerson().getName());
 
-            LocalizedString body =
-                    BundleUtil.getLocalizedString(Bundle.MESSAGING, "thesis.announcement.body", thesis.getStudent().getPerson()
-                            .getName(), getDate(thesis.getProposedDiscussed()), String.valueOf(hasPlace(thesis)),
-                            thesis.getProposedPlace(), String.valueOf(hasTime(thesis.getProposedDiscussed())),
-                            getTime(thesis.getProposedDiscussed()), thesis.getTitle().getContent());
+            LocalizedString bodyFormat = BundleUtil.getLocalizedString(Bundle.MESSAGING, "thesis.announcement.body");
+            String bodyString = MessageFormat.format(bodyFormat.getContent(), thesis.getStudent().getPerson().getName(),
```

Comment:
By using the `getContent()` method you are losing locale information, as you're only processing the string for the current locale of the user, only to then pass it back into a LocalizedString.

I suggest you take a look at LocalizedString's `map` method.